### PR TITLE
use contants in jdl/validators folder

### PR DESCRIPTION
**Pull request by renanfranca**
contributes to #14235

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by mshima — COMMENTED** [review]
Please extract builtInEntities into a common place, maybe @Tcharl has a suggestion where to put or if it already exists..

File: jdl/validators/jdl-with-application-validator.js [hunk 1]
```diff
@@ -107,7 +111,7 @@ function createValidator(jdlObject, logger = console, options = {}) {
   function checkForFieldErrors(entityName, jdlFields, jdlApplication) {
     const validator = new FieldValidator();
     const filtering =
-      jdlApplication.getConfigurationOptionValue('databaseType') === 'sql' &&
+      jdlApplication.getConfigurationOptionValue('databaseType') === DatabaseTypes.SQL &&
```
Comment (Tcharl) [inline jdl/validators/jdl-with-application-validator.js:114]: Destructuring like everywhere els?

File: jdl/validators/jdl-without-application-validator.js [hunk 2]
```diff
@@ -202,7 +206,7 @@ function checkForAbsentEntities({ jdlRelationship, doesEntityExist, skippedUserM
   }
 }
 function isUserManagementEntity(entityName) {
-  return entityName.toLowerCase() === 'user' || entityName.toLowerCase() === 'authority';
+  return builtInEntities.has(entityName.toLowerCase());
```
Comment (Tcharl) [inline jdl/validators/jdl-without-application-validator.js:209]: For now this one works, but semantically speaking this change is a little bit risky: WDYT?

**Comment by Tcharl** [comment]
for builtInEntities, they usually should go in jdl/jhipster/built-in-entities.js, shouldn't they?

File: jdl/validators/jdl-without-application-validator.js (hunk 2 above)
Comment (renanfranca) [reply jdl/validators/jdl-without-application-validator.js:209]: I could undo this. I look up this approach from the Jhipster codebase. I don't if this is better or worse..

File: jdl/validators/jdl-with-application-validator.js (hunk 1 above)
Comment (renanfranca) [reply jdl/validators/jdl-with-application-validator.js:114]: I don't understand. Should i use === SQL instead of DatabaseTypes.SQL?

Comment (Tcharl) [reply jdl/validators/jdl-with-application-validator.js:114]: Yes, totally
```
import { SQL } = require('../../jdl/jhipster/database-types');
...
jdlApplication.getConfigurationOptionValue('databaseType') === SQL && ...
```

**Pull request closed without merging**